src/NuGet.Clients/PackageManagement.UI/Models/SearchResultPackageMetadata.cs

```diff
@@ -4,41 +4,297 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Globalization;
 using System.Threading.Tasks;
-using NuGet.Protocol.Core.Types;
+using System.Windows;
 using NuGet.Protocol.VisualStudio;
 using NuGet.Versioning;
 
 namespace NuGet.PackageManagement.UI
 {
+    // This is the model class behind the package items in the infinite scroll list.
+    // Some of its properties, such as Latest Version, Status, are fetched on-demand in the background.
     public class SearchResultPackageMetadata : INotifyPropertyChanged
     {
+        public SearchResultPackageMetadata()
+        {
+            InstalledVersionIndicatorVisibility = Visibility.Collapsed;
+        }
+
         private PackageStatus _status;
+
         public event PropertyChangedEventHandler PropertyChanged;
 
+        private static readonly string[] _scalingFactor = new string[] {
+            "",
+            "K", // kilo
+            "M", // mega, million
+            "G", // giga, billion
+            "T"  // tera, trillion
+        };
+
         public string Id { get; set; }
 
+        // Indicates whether the instance is created for solution package manager
+        public bool IsSolution { get; set; }
+
+        // Indicates whether the instance is created for the update tab.
+        public bool IsUpdateTab { get; set; }
+
+        public string Author { get; set; }
+
         public NuGetVersion Version { get; set; }
 
+        private NuGetVersion _installedVersion;
+
+        // The installed version of the package.
+        public NuGetVersion InstalledVersion
+        {
+            get
+            {
+                return _installedVersion;
+            }
+            set
+            {
+                if (!VersionEquals(_installedVersion, value))
+                {
+                    _installedVersion = value;
+                    OnPropertyChanged(nameof(InstalledVersion));
+
+                    // update tool tip
+                    if (_installedVersion != null)
+                    {
+                        var versionForDisplay = new VersionForDisplay(_installedVersion, string.Empty);
+                        InstalledVersionToolTip = string.Format(
+                            CultureInfo.CurrentCulture,
+                            Resources.ToolTip_InstalledVersion,
+                            versionForDisplay);
+                    }
+                    else
+                    {
+                        InstalledVersionToolTip = null;
+                    }
+
+                    // update installed version indicator visibility
+                    if (_installedVersion != null && !IsSolution)
+                    {
+                        InstalledVersionIndicatorVisibility = Visibility.Visible;
```

Comment:
you should use triggers for this, rather than the view model, not a huge deal though